### DISCUSSION TITLE
Cant't run sunshine after updating the system

### DISCUSSION THREAD

**Discussion by JamHadar** [LizardByte/community#313] [Sunshine] [answered]
A few days ago I tried to run Sunshine and got this error:

`sunshine: error while loading shared libraries: libboost_locale.so.1.81.0: cannot open shared object file: No such file or directory`

I'm using Arch and Nvidia. Installed via AUR.

I assume it is because boost-libs package recently updated and it is now 1.83. I have few thoughts how to approach this problem, but I'm still new to linux and don't want to bork my system.
1. Is there a way to edit config or smth to make Sunshine use current libraries?
2. As far as I know, Flatpak is a sandbox. If I install Sunshine via Flatpak, will Sunshine run with older libs?

**Comment by KuleRucket** [comment, marked as answer]
See #1600

The library got updated by Arch after AUR built against it. You just need to reinstall from AUR to force a rebuild.

**Reply by JamHadar** [reply]
Does this mean it will use current libs and not reinstall old ones?

**Reply by ReenigneArcher** [reply]
Generally speaking, Sunshine links against your libraries, it doesn't install them.

**Reply by JamHadar** [reply]
Got it. Thanks

**Reply by maddox** [reply]
> You just need to reinstall from AUR to force a rebuild

I'm not sure I understand here. Are we supposed to reinstall sunshine to solve this? I have tried that and it still won't work. As it's still trying to use `1.81.0`.

What exactly do we need to reinstall to force a rebuild so Sunshine is pointing at `1.83.0`?

**Reply by JamHadar** [reply]
I did `makepkg -fsi` to force rebuild. As I understand, you need to rebuild it first, then install with new libs.

**Reply by Pyro57000** [reply]
When I git clone from the AUR (`git clone https://aur.archlinux.org/sunshine`) then try to run makepkg -fsi inside the directory I get the following error:

`/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp: In member function ‘bool upnp::deinit_t::create_ipv6_pinholes()’:
/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp:112:37: error: too few arguments to function ‘int UPNP_GetValidIGD(UPNPDev*, UPNPUrls*, IGDdatas*, char*, int, char*, int)’
  112 |       auto status = UPNP_GetValidIGD(device.get(), &urls.el, &data, lan_addr.data(), lan_addr.size());
      |                     ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp:5:
/usr/include/miniupnpc/miniupnpc.h:122:1: note: declared here
  122 | UPNP_GetValidIGD(struct UPNPDev * devlist,
      | ^~~~~~~~~~~~~~~~
/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp: In member function ‘void upnp::deinit_t::upnp_thread_proc()’:
/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp:334:39: error: too few arguments to function ‘int UPNP_GetValidIGD(UPNPDev*, UPNPUrls*, IGDdatas*, char*, int, char*, int)’
  334 |         auto status = UPNP_GetValidIGD(device.get(), &urls.el, &data, lan_addr.data(), lan_addr.size());
      |                       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/miniupnpc/miniupnpc.h:122:1: note: declared here
  122 | UPNP_GetValidIGD(struct UPNPDev * devlist,
      | ^~~~~~~~~~~~~~~~
/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp: At global scope:
/home/pyro/git-projects/sunshine/src/sunshine/src/upnp.cpp:47:3: error: ‘std::string_view upnp::status_string(int)’ defined but not used [-Werror=unused-function]
   47 |   status_string(int status) {
      |   ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/sunshine.dir/build.make:104: CMakeFiles/sunshine.dir/src/upnp.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:191: CMakeFiles/sunshine.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
make: Leaving directory '/home/pyro/git-projects/sunshine/src/build'
==> ERROR: A failure occurred in build().
    Aborting...`

**Reply by ReenigneArcher** [reply]
@Pyro57000 fixed by https://github.com/LizardByte/Sunshine/pull/2782

**Comment by KitsuneNoNatsu** [comment]
Hey, I'm still getting this issue even now, I installed Sunshine today

**Reply by ReenigneArcher** [reply]
@KitsuneNoNatsu fixed by #2782

sunshine AUR is not pointing a beta version of Sunshine though so you will need to install it a different way.

**Comment by theKasten** [comment]
Having the same issue when installing through yay. Same issue when downloading manually from aur (and running makepkg).

"error: too few arguments to function ‘int UPNP_GetValidIGD"
